Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -18,51 +18,32 @@ class CustomerInfoResponseHandler {
     init() { }
 
     func handle(customerInfoResponse response: Result<HTTPResponse, Error>,
-                file: String = #fileID,
-                function: String = #function,
-                line: UInt = #line,
                 completion: BackendCustomerInfoResponseHandler) {
-        switch response {
-        case let .failure(error):
-            completion(.failure(
-                ErrorUtils.networkError(withUnderlyingError: error,
-                                        fileName: file, functionName: function, line: line)
-            ))
-
-        case let .success(response):
-            let (statusCode, response) = (response.statusCode, response.jsonObject)
-            let isErrorStatusCode = !statusCode.isSuccessfulResponse
-
-            let errorResponse = ErrorResponse.from(response)
-
-            var result: Result<CustomerInfo, Error> = {
-                // Only attempt to parse a response if we don't have an error status code from the backend.
-                if !isErrorStatusCode {
-                    return Result { try CustomerInfo.from(json: response) }
+        let errorResponse = ErrorResponse.from(response.value?.jsonObject ?? [:])
+
+        let result: Result<CustomerInfo, Error> = response
```

Comment:
Visually, this is more confusing to me. Is there a way to structure this so it's easier to parse? Maybe I'm the only one, but that was something I struggled with learning RxSwift/ReactiveCocoa. The many `.flatMap` after `.flatMap` with more nested function calls like `mapError` it's difficult for me to follow the flow of logic/transformations.

Reply:
Yeah I brought up a similar comment [here](https://github.com/RevenueCat/purchases-ios/pull/1425#discussion_r839915462).
For this particular example this code is going away in that PR since the deserialization will be done by `HTTPClient`, so this is just temporary.

But the point is still valid, especially in that other PR.

Reply:
+1 on the functional code being tougher to understand. We can continue discussion on the other PR if this is temporary

Reply:
Yeah this is gone later since all the parsing will be done by `HTTPClient`, but welcoming feedback on the other one.